jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/TransferUtils.kt

```diff
@@ -75,18 +75,22 @@ fun S3Client.download(
     destination: Path,
     message: String = message("s3.download.object.progress", key),
     startInBackground: Boolean = true
-): CompletionStage<GetObjectResponse> = download(project, bucket, key, destination.outputStream(), message, startInBackground)
+): CompletionStage<GetObjectResponse> = download(
+    project,
+    GetObjectRequest.builder().bucket(bucket).key(key).build(),
+    destination.outputStream(),
+    message,
+    startInBackground
+)
 
 fun S3Client.download(
     project: Project,
-    bucket: String,
-    key: String,
+    request: GetObjectRequest,
     destination: OutputStream,
-    message: String = message("s3.download.object.progress", key),
+    message: String = message("s3.download.object.progress", request.key()),
```

Comment:
`key()` can be null since the `GetObjectRequest` doesn't enforce that the key be set

Reply:
No longer `GetObjectRequest` in parameter here guided by other suggestions :+1: